File: SecondYear/sauban123_MohdSauban_2226csit1130_2/Week_4/MaximumConsecutiveOnes.cpp
int traffic(int n, int m, vector<int> vehicle) {
	// Write your code here.

	int i=0;

int k=0;

while(i<n){

if(vehicle[i]==0){

m--;

}if(m<=-1){

if(vehicle[k]==0){

m++;

}k++;

}i++;

}return i-k;
	
}

File: SecondYear/aayushraghav93_AayushRaghav_2226CSE(AI)1108_2/Week_4/maximumconsecutiveones.java
public class Solution {
    public static int traffic(int n, int m, int []vehicle) {
        // Write your code here.
        int sum = 0;
    int c = 0;
    int j = 0;

    for (int i = 0; i < n; i++) {
        while (j < n && c <= m) {
            if (vehicle[j] == 0) {
                c++;
                if (c > m) {
                    c--;
                    break;
                }
            }
            j++;
        }
       sum = Math.max(sum, j - i);
        if (vehicle[i] == 0) {
            c--;
        }
    }

    return sum;
    }
}
